### DISCUSSION TITLE
Linechart tootltip data is static When a Notebook is Embedded

### DISCUSSION THREAD

**Discussion by mcollie1** [observablehq/plot#1065] [Q&A]
I've used the `addTooltip` function in a notebook. While in the Notebook directly, data updates as one hovers over different points a the line. However, data becomes static when embedding the notebook.

https://observablehq.com/@mkfreeman/plot-tooltip

**Comment by harshagarwalnyu** [comment]
The `addTooltip` community notebook uses custom pointer-event listeners attached to the document that don't transfer correctly into cross-origin iframes (embedded notebooks). The tooltip gets initialized but the hover events never fire in the embedding context.

**Fix: use Plot's built-in `tip` mark** (added in Plot 0.6.7) instead of the community `addTooltip` function:

```js
Plot.plot({
  marks: [
    Plot.line(data, {x: "date", y: "value"}),
    Plot.tip(data, Plot.pointerX({x: "date", y: "value"}))
  ]
})
```

The native `tip` mark attaches events directly to the SVG element rather than the `document`, so it works correctly in iframes and embedded contexts.

**If you must keep `addTooltip`:** ensure the embedded iframe has `allow="pointer-events"` and the host page's CSP permits it. But the built-in `tip` mark is the correct long-term solution — `addTooltip` was a community workaround before the `tip` mark existed.
